robots/pkg/flakefinder/report_data.go

```diff
@@ -211,16 +211,29 @@ func SortTestsByRelevance(data map[string]map[string]*Details, tests []string) (
 			}
 			flakinessToTestNames[details.Severity] = append(flakinessToTestNames[details.Severity], test)
 
-			// if we only count the number of occurrences a test failed then tests with lower number of failures but same number of
-			// appearances might appear before tests that have higher number of failures.
+			// Counting the occurrences a test failed will cause tests with lower number of failures but
+			// same number of appearances to appear before tests that have higher number of failures.
+			// Also using the ratio of #tests_failed / #tests_succeeded as a factor emphasizes tests
+			// with less failures but a high ratio far more than tests with higher number of tests.
 			// As we want to emphasize tests that have higher failure numbers over those that have less failures,
-			// we calculate the severity ratio as #tests_failed / #tests_succeeded, which gives us a basic ratio and
-			// then multiply that value by the number of failures. This yields:
-			severityRatio := 1 + int(float64(details.Failed*details.Failed)/float64(details.Succeeded+1))
+			// we tabularize the severity as a factor according to the relation of #tests_failed / #tests_succeeded
+			// We then multiply that value by the number of failures.
+			var severityRatio float32
+			switch {
+			case details.Failed > 3*details.Succeeded:
```

Comment:
moving to a severityRatio set to steps of {1,0.75,1.5,1.75} will cause these cases to have the same severityRatio, is that what we want?
case 1 (just above the threshold) : details.Failed > 3 * details.Succeeded+1
case 2 (way above the threshold): details.Failed > 10 * details.Succeeded

Reply:
@RamLavi can you expand a bit please? I can't follow.

I am trying to emphasize tests with many failures far more than tests with few failures, thus I avoid using a proper ratio between the two values. Please also see the test cases that were added: https://github.com/kubevirt/project-infra/pull/2095/files#diff-430ce2e6d483c0ab365ca194a0711454599b2fb9e1406b5f17f76d53d897f941R158

Reply:
> moving to a severityRatio set to steps of {1,0.75,1.5,1.75} will cause these cases to have the same severityRatio, is that what we want?
> case 1 (just above the threshold) : details.Failed > 3 * details.Succeeded+1
> case 2 (way above the threshold): details.Failed > 10 * details.Succeeded

In the end the number of failures should be much more important than the ratio of failures to succeeding test cases. As in l232 the severity is multiplied with the ratio, this rough number of classes for ratio would be less emphasizing than what we were doing before.

Maybe sth like 

```
0.75 + (details.Failed) / 4 * details.Succeeded)
```

would be better? This way the emphasis of ratio would be much smaller, while the ratio of failures vs. succeeding tests would still be expressed?

@RamLavi WDYT?

Reply:
@enp0s3 @RamLavi [this would be an alternative](https://github.com/dhiller/project-infra/commit/5a4cf1e8978a7542ef67fb53b2e51a051f6a1d68), but then I need to live with one case not working (used Ginkgo Pending for that)